src/metabase/models/database.clj

```diff
@@ -276,6 +276,8 @@
                  (or
                   ;if there is any changes in user control setting
                   (some? (get-in changes [:details :let-user-control-scheduling]))
+                  ;; if the let user control scheduling is already on, we should always try to re-infer it
+                  (get-in database [:details :let-user-control-scheduling])
```

Comment:
Why go for always recalculating in this case, versus watching for changes on `is_full_sync` or `is_on_demand`? It feels slightly lopsided to me to mix logic on changes and current state. That said, having false positives where we infer the same schedule again seems fine since we don't randomize in this case.

Reply:
Yeah, that works, too, but I plan to tackle this soon: https://github.com/metabase/metabase/issues/41345, and all of those flags will go away.

for now I'll keep this because I don't want to wait for CI again :(